lms/djangoapps/commerce/signals.py

```diff
@@ -0,0 +1,107 @@
+"""
+Signal handling functions for use with external commerce service.
+"""
+import logging
+
+from django.dispatch import receiver
+from ecommerce_api_client.exceptions import HttpClientError
+from request_cache.middleware import RequestCache
+from student.models import UNENROLL_DONE
+
+from commerce import ecommerce_api_client, is_commerce_service_configured
+
+log = logging.getLogger(__name__)
+
+
+@receiver(UNENROLL_DONE)
+def handle_unenroll_done(sender, course_enrollment=None, skip_refund=False, **kwargs):  # pylint: disable=unused-argument
+    """
+    Signal receiver for unenrollments, used to automatically initiate refunds
+    when applicable.
+
+    N.B. this signal is also consumed by lms.djangoapps.shoppingcart.
+    """
+    if not is_commerce_service_configured():
+        return
+
+    if course_enrollment and course_enrollment.refundable() and not skip_refund:
```

Comment:
Minor: move `not skip_refund` to the beginning in case `course_enrollment.refundable()` is a resource hog (which can be bypassed in the event `skip_refund` is `True`).